### PR TITLE
Implemented Class Decoration

### PR DESCRIPTION
**Pull request by Rana-Mudassir**
In this PR following tasks are done.

- [x] Implement composition (as an example of the design pattern).
- [x] Create a class Nameable for Interface
- [x] Turn your Person class to Nameable
- [x] Prepare base Decorator
- [x] Prepare CapitalizeDecorator and TrimmerDecorator

Thanks for Reviewing 😊

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+class Person < Nameable
+  def initialize(name, age = 'Unknown', parent_permission: true)
+    super()
+    @id = Random.rand(1..1000)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  def correct_name
```
Comment (rustam-tolipov) [inline person.rb:14]: **You have done great job!**

- [ ] However, your based on requirement testing if the app is working as expected is not completely held up. When I run the test in `person.rb`:

```ruby
person = Person.new(22, 'maximilianus')
  person.correct_name
  capitalizedPerson = CapitalizeDecorator.new(person)
  capitalizedPerson.correct_name
  capitalizedTrimmedPerson = TrimmerDecorator.new(capitalizedPerson)
  capitalizedTrimmedPerson.correct_name
``` 

it shows me errors. Kindly fix the issue and resubmit your app. If you have any question please reach me out via slack

**Review by rustam-tolipov — CHANGES_REQUESTED** [review]
Hello @Rana-Mudassir,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
_Your project is almost complete!_
_No Linters errors_
_Well-implemented functions_
_Great readme_

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Check the comments under the review._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (Rana-Mudassir) [reply person.rb:14]: Thanks for reviewing. Error is resolved now

**Review by Kossi-stack — APPROVED** [review]
## Status: Approved ✔️

**Hi @Rana-Mudassir,**

You did a really great job! 💯 💯💯💯
Congratulations! 🎉 🎉 🎉 🎉 

128382030-38e7de91-f6f2-45cf-b54e-8068e0446e4c

You may close this Pull Request and merge the changes
Keep being awesome dev's and good luck with your next milestone!

## To Highlight

✅ Descriptive pull request message
✅ Linter error checks validated
✅ Documented your work professionally
✅ README file is descriptive.
✅ Commit messages are meaningful.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers, and Happy coding! 👏 👏 👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

**Pull request merged**